src/utils.c

```diff
@@ -370,6 +370,11 @@ gboolean r_read_exact(const int fd, guint8 *data, size_t size, GError **error)
 					"Failed to read: %s", g_strerror(err));
 			return FALSE;
 		} else if (ret == 0) { /* end of file */
+			int err = errno;
```

Comment:
`errno` doesn't contain anything useful when pread returns 0, so we shouldn't try to add that to the error string.

Reply:
@jungte this still remains as an unsued variable in your patch now. If you fix this also (and the occurrence noted below) than the PR should be ready for being merged.

Reply:
@ejoerns OK, how about this?